worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,223 @@
+    multiworld.get_region('6-8', player).add_exits(["Bowser's Room"],{"Bowser's Room": lambda state: bosses.castle_clear(state)})
+
+    if world.options.extras_enabled == 1:
+        multiworld.get_region('World 1', player).add_exits(['1-Extra'],
+            {'1-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 1'}, player)})
+        multiworld.get_region('World 2', player).add_exits(['2-Extra'],
+            {'2-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 2'}, player)})
+        multiworld.get_region('World 3', player).add_exits(['3-Extra'],
+            {'3-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 3'}, player)})
+        multiworld.get_region('World 4', player).add_exits(['4-Extra'],
+            {'4-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 4'}, player)})
+        multiworld.get_region('World 5', player).add_exits(['5-Extra'],
+            {'5-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 5'}, player)})
+        multiworld.get_region('World 6', player).add_exits(['6-Extra'],
+            {'6-Extra': lambda state: state.has_any({'Extra Panels', 'Extra 6'}, player)})
+
+    if world.options.minigame_checks.value >= 2:
+        multiworld.get_region('World 1', player).add_exits(['1-Bonus'],
+            {'1-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 1'}, player)})
+        multiworld.get_region('World 2', player).add_exits(['2-Bonus'],
+            {'2-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 2'}, player)})
+        multiworld.get_region('World 3', player).add_exits(['3-Bonus'],
+            {'3-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 3'}, player)})
+        multiworld.get_region('World 4', player).add_exits(['4-Bonus'],
+            {'4-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 4'}, player)})
+        multiworld.get_region('World 5', player).add_exits(['5-Bonus'],
+            {'5-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 5'}, player)})
+        multiworld.get_region('World 6', player).add_exits(['6-Bonus'],
+            {'6-Bonus': lambda state: state.has_any({'Bonus Panels', 'Bonus 6'}, player)})
+
+def create_location(player: int, location_data: LocationData, region: Region, location_cache: List[Location]) -> Location:
+    location = YILocation(player, location_data.name, location_data.code, region)
+    location.access_rule = location_data.rule
+    location.LevelID = location_data.LevelID
+
+    if id is None:
+        location.event = True
+        location.locked = True
+
+    location_cache.append(location)
+
+    return location
+
+def create_region(multiworld: MultiWorld, player: int, locations_per_region: Dict[str, List[LocationData]], location_cache: List[Location], name: str) -> Region:
+    region = Region(name, player, multiworld)
+    region.world = multiworld

```

Comment:
If we're gonna take Treble's suggestion:
```suggestion
def create_region(world: "YIWorld", player: int, locations_per_region: Dict[str, List[LocationData]], location_cache: List[Location], name: str) -> Region:
    region = Region(name, player, world.multiworld)

```
(I'm also not sure what the purpose of `region.world = multiworld` is, since Region objects don't have a `world` attr, and a `multiworld` is not a `world`)